FOUR-22600: Issues with stages

pmPaulis opened the pull request:
## Issue & Reproduction Steps
There is confusion with the stages when they are not published.
Error in alternatives when changing the position of the stages

## Solution
- List the changes you've introduced to solve the issue.

## How to Test
Describe how to test that this solution works.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-25926
- https://processmaker.atlassian.net/browse/FOUR-25905
- https://processmaker.atlassian.net/browse/FOUR-25979

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy

henryjonathanquispe committed:
FOUR-25905: Error in alternatives when changing the position of the stages

henryjonathanquispe committed:
FOUR-25905: Error in alternatives when changing the position of the stages

henryjonathanquispe committed:
FOUR-25905: Error in alternatives when changing the position of the stages

gproly committed:
FOUR-23421 There is confusion with the stages when they are not published.
 ## Description:
Create a process
Add Start Event → Task Form → End Event
Create two stages and assigned these to flow to each task
Publish the process
Open the process by launchpad
Create a case
Return to process
Delete the stages in the process
Does not publish the changes
Go to process launchpad
Check that stages are not present after the changes
Create new cases
The new cases are created with stage but it is not visible in process launchpad
Current Behavior:
To stages are visible to start a CASE is necessary to publish the process with stages
To stages are visible in LAUNCH PAD does not necessary to publish the process with stages
which causes confusion when displaying case information
Expected Behavior:
The information in launch pad should be sync
If there are stages the new cases should be created with the corresponding stage
If there are not stages the new cases should be created without stages
To this it necessary that stages will be publish to avoid errors and confusions

 ## Related Tickets & Packages
https://processmaker.atlassian.net/browse/FOUR-25926

gproly committed:
FOUR-25926 Observations

gproly committed:
FOUR-25926 Fix the observations.

pmPaulis committed:
Merge pull request #8453 from ProcessMaker/bugfix/FOUR-25905

FOUR-25905: Error in alternatives when changing the position of the stages

pmPaulis committed:
Merge pull request #8474 from ProcessMaker/bugfix/FOUR-25926

FOUR-23421 There is confusion with the stages when they are not publi…

vladyrichter commented:
QA server K8S was successfully deployed https://ci-e99bdcbf88.engk8s.processmaker.net

gproly committed:
FOUR-25979 Label are not lost in alternative B

 ## Description
- Steps to Reproduce
Create a process
Add Start Event → Task form → End Event
Add new stages
Assign the stages in the flow
Create alternative B
Delete one alternative
Publish the changes
- Current Behavior
Stage labels are not lost in both alternatives, this remain in alternative B
- Expected Behavior
Stage labels should lost in both alternatives

 ## Related Tickets & Packages
https://processmaker.atlassian.net/browse/FOUR-25979

gproly committed:
FOUR-25979 Remove unnecessary code.

pmPaulis committed:
Merge pull request #8499 from ProcessMaker/bugfix/FOUR-25979

FOUR-25979 Label are not lost in alternative B
